invenio_accounts/api.py

```diff
@@ -8,25 +8,60 @@
 
 """API objects for Invenio Accounts."""
 
-from collections import defaultdict
+
+class Session:
+    """Session object for DB Users change history."""
+
+    def __init__(self):
+        self.updated_users = []
+        self.updated_roles = []
+        self.deleted_users = []
+        self.deleted_roles = []
+        self.indexed = False
```

Comment:
In the demo it was shown that the attributes are updated from invenio-access and invenio-communities. I would suggest having methods inside the `Session` class e.g.:

```py
def index():
     self.indexed = True
```

To encapsulate the handling of the flags. If we update from other modules we might end up in a situation where we do not know exactly who is updating the flags. E.g. avoid deadlocks.

Reply:
Good idea! thanks :+1: